Scripts/RemediationScripts/Remediate-SetStorageAccountMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,959 @@
+       
+            $totalstorageAccountResources = ($storageAccountResources | Measure-Object).Count
+        
+        }
+        else
+        {
+            if (-not (Test-Path -Path $FilePath))
+            {
+                Write-Host "ERROR: Input file - $($FilePath) not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host $([Constants]::DoubleDashLine)
+                break
+            }
+
+            Write-Host "Fetching all Storage Accounts(s) from $($FilePath)" -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            $storageAccountResourcesFromFile = Import-Csv -LiteralPath $FilePath
+            $validstorageAccountResources = $storageAccountResourcesFromFile | Where-Object { ![String]::IsNullOrWhiteSpace($_.StorageAccountName) }
+
+            $validstorageAccountResources | ForEach-Object {
+                $resourceGroupName = $_.ResourceGroupName        
+                $storageAccountName = $_.StorageAccountName               
+                try
+                {
+                    $storageAccount = Get-AzStorageAccount  -ResourceGroupName $resourceGroupName -Name $storageAccountName -ErrorAction SilentlyContinue
+                    $storageAccountResources += $storageAccount | Select-Object @{N='StorageAccountName';E={$_.StorageAccountName}},
+                                                                @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                                @{N='PrimaryLocation';E={$_.PrimaryLocation}},
+                                                                @{Name ='MinimalTlsVersion';Expression={ (Get-AzStorageAccount -Name $_.StorageAccountName -ResourceGroupName $_.ResourceGroupName).MinimumTlsVersion }}
+
+                }
+                catch
+                {
+                    Write-Host "Error fetching Storage Account:   - $($StorageAccountName). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                    Write-Host "Skipping this Storage Account..." -ForegroundColor $([Constants]::MessageType.Warning)
+                }
+            }
+        }
+    }
+    
+    $totalstorageAccountResources = ($storageAccountResources | Measure-Object).Count
+
+    if ($totalstorageAccountResources -eq 0)
+    {
+        Write-Host "No Storage Accounts found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)
+        break
+    }
+
+    Write-Host "Found $($totalstorageAccountResources) Storage Account(s)." -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine)
+ 
+     
+    # Includes Storage Accounts where minimal required TLS version is set  
+    $StorageAccountsWithReqMinTLSVersion = @()
```

Comment:
Why have we defined this? Is there any use?

Reply:
Not required actually. Removed